lib/sinon.js

```diff
@@ -7,7 +7,21 @@
  * Copyright (c) 2010-2013 Christian Johansen
  */
 "use strict";
-module.exports = require("./sinon/util/core");
+
+var match = require("./sinon/match");
+
+module.exports = exports = require("./sinon/util/core");
+
+exports.extend = require("./sinon/extend");
+exports.match = match;
+exports.typeOf = require("./sinon/typeOf");
+
+/*
+ * allow deepEqual to check equality of matchers through
+ * dependency injectection. Otherwise we get a circular
+ * dependency
+ */
+exports.deepEqual.match = match;
```

Comment:
Dependency injection is usually done via the arguments of a function. In this pattern, `deepEqual` will have different behaviour if it's extended by this file or not.

```javascript
function deepEqual(match, a, b) {
```

I would much prefer using an extra argument, if that mean solving circular dependency problems